materialize-sql/validate.go

```diff
@@ -124,12 +127,11 @@ func ValidateMatchesExisting(resource Resource, existing *pf.MaterializationSpec
 }
 
 func checkTypeError(field string, existing *pf.CollectionSpec, proposed *pf.CollectionSpec) string {
-	// existingProjection is guaranteed to exist since the MaterializationSpec has already been
-	// validated.
 	var existingProjection = existing.GetProjection(field)
 	var proposedProjection = proposed.GetProjection(field)
-	if proposedProjection == nil {
```

Comment:
If I'm following this correctly, we don't need the `proposedProjection == nil` check here because of the change in this PR & specifically the addition on line 103-105 means we won't even call `checkTypeError` when removing a field.

I'm trying to grok the addition of `existingProjection == nil` below. Since `checkTypeError` is being called out of a loop on `existing.FieldSelection.AllFields()` I would think that `existingProjection` would never be `nil`?